core/templates/dev/head/collection_editor/CollectionNodeDirective.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for editing a collection node. This includes:
+ * editing pre-requisite skills, editing acquired skills and deleting a node.
+ *
+ * @author mgowano@google.com (Abraham Mgowano)
+ */
+
+oppia.directive('collectionNodeDirective', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      getPrerequisiteSkills: '&prerequisiteSkills',
+      getAcquiredSkills: '&acquiredSkills',
+      getExplorationId: '&explorationId',
+      getExplorationTitle: '&explorationTitle',
+      addChange: '&'
+    },
+    templateUrl: 'inline/collection_node_directive',
+    controller: ['$scope', 'CollectionUpdateService', function($scope,
+      CollectionUpdateService) {
+      var _arrayContainsCaseInsensitive = function(array, element) {
+        var lowerElement = element.toLowerCase();
+        for (var i = 0; i < array.length; i++) {
+          if (array[i].toLowerCase() === lowerElement) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+      $scope.addPrereqSkill = function(skillName) {
+        if (skillName) {
+          if (!_arrayContainsCaseInsensitive($scope.getPrerequisiteSkills(), skillName)) {
+            $scope.getPrerequisiteSkills().push(skillName);
```

Comment:
I think the miscommunication in our last meeting was with regards to whether the controller should create the change object which is sent to the backend. I think the directive should create the change object and add it to the change list stored in the controller (exactly as it's done here), but the action of adding a prerequisite skill should probably be done by calling an ``addPrerequisiteSkill()`` provided by the controller. There are three reasons for this:

1. It keeps the overall, complex logic needed to update the prerequisite skill of the collection out of the controller.
2. Using a function keeps ``$scope.getPrerequisiteSkills()`` read-only, which I think it should since it will be very hard for someone working on the controller to find places where the list is modified.
3. We will eventually be introducing a ``Collection`` object which will contain functions like ``getPrerequisiteSkills()`` and ``addPrerequisiteSkill()`` (similar to collection_domain.Collection), so that will further thin out the controller and keep the directive roughly the same.

For those reasons I lean towards the suggested change proposed.